lib/hypervisor/hv_kvm/monitor.py

```diff
@@ -553,10 +553,8 @@ def HotDelDisk(self, devid):
     """
     self.execute_qmp("device_del", {"id": devid})
 
-    # TODO: implement receiving of QMP events
-    # We need to wait for the DEVICE_DELETED event via QMP before proceeding.
-    # The old implementation using HMP only worked because it is so slow.
-    time.sleep(1)
+    # wait for the DEVICE_DELETED event with one minute timeout
```

Comment:
I assume this should be one second not minute?

Reply:
I think a minute is a good choice here. [Qemu docs](https://qemu.readthedocs.io/en/v8.1.5/interop/qemu-qmp-ref.html#qapidoc-2461) state:

> Hot removal is an operation that requires guest cooperation. This command merely requests that the guest begin the hot removal process. Completion of the device removal process is signaled with a DEVICE_DELETED event.

It does not really say what is going on inside the guest and how long it might take to complete but I would assume that a single second _could_ cause problems.

While our experience tells us that a second has been enough in the past (either because the `socat` / HMP implementation was too slow or because of the explicit `time.sleep(1)`) I would not bet on it to be accurate in _all_ cases :-)

Reply:
A one-minute timeout is quite a lot for me. If something unexpected happens, the execution is blocked for a whole minute. I would suggest 5-10 seconds as timeout.

Reply:
I have asked on the OFTC #qemu IRC channel. Let's see if they have any advice on this. If this is about (potentially) flushing buffers/caches or whatnot, we should probably give the guest some time to deal with that.

If it is just about "has the guest OS support for hot removal or not", a short timeout should be sufficient I guess.